src/components/Card/Card.js

```diff
@@ -15,6 +15,16 @@ function Card(props) {
   const [editedTitle, setEditedTitle] = useState(props.title);
   const [editedDescr, setEditedDescr] = useState(props.descr);
 
+  // Функция для сброса режима редактирования
+  const resetEditingMode = () => {
+    setIsEditing(false);
+  };
+
+  // Проверка на изменение props.readOnly и сброс режима редактирования при необходимости
+  if (props.readOnly && isEditing) {
+    resetEditingMode();
+  }
```

Comment:
Try to do this using useEffect hook.
What is the difference between your solution and useEffect?

Reply:
In my solution, the code is executed every time the component is rendered. If I use the useEffect hook, it will execute this code only when the value of the dependencies is changed.